### DISCUSSION TITLE
"Compilation failed" when running RAILS_ENV=production bundle exec rails assets:precompile

### DISCUSSION THREAD

**Discussion by mtbluedog** [mastodon/mastodon#23171] [Q&A] [answered]
When I run 
`Mastodon:~/live$ RAILS_ENV=production bundle exec rails assets:precompile`
the output is:
```
yarn install v1.22.19
[1/6] Validating package.json...
[2/6] Resolving packages...
success Already up-to-date.
Done in 2.30s.
Compiling...
Compilation failed:
```
Any ideas why I see "Compilation failed"? Are there logs that will show the reason for the failure? Or can I ignore that warning?

**Comment by ClearlyClaire** [comment, marked as answer]
This can happen when `webpack` is running out of memory, you can try to add swap, or reduce build parallelism:
```patch
diff --git a/config/webpack/production.js b/config/webpack/production.js
index 143a23b99..f53a1b3a4 100644
--- a/config/webpack/production.js
+++ b/config/webpack/production.js
@@ -13,6 +13,7 @@ const sharedConfig = require('./shared');
 const root = resolve(__dirname, '..', '..');
 
 module.exports = merge(sharedConfig, {
+  parallelism: 1,
   mode: 'production',
   devtool: 'source-map',
   stats: 'normal',
```

Due to webpacker's inadequate handling of such build failures, you will likely need to run `RAILS_ENV=production bundle exec rails assets:clobber` before attempting again, as webpacker may pretend there is nothing to do even though it did fail to build the assets.

**Reply by mtbluedog** [reply]
Thanks! That worked. in my DO Droplet, I doubled CPU and RAM to 2CPU and 4GB and added the parallelism parameter, and now compiling doesn't fail. I also found I needed to run clobber and force webpacker to rebuild.
